docs/source/howto/workflows.rst

```diff
@@ -135,171 +293,170 @@ The second argument is the result of the work chain, extracted from the ``Int``
-
-If you are unfamiliar with the inputs of a particular ``WorkChain``, a convenient tool for setting up the work chain is the :ref:`process builder<topics:processes:usage:builder>`.
-This can be obtained by using the ``get_builder()`` method, which is implemented for every ``CalcJob`` and ``WorkChain``:
-
-.. code-block:: ipython
-
-    In [1]: from aiida.plugins import WorkflowFactory, DataFactory
-       ...: Int = DataFactory('int')
-       ...: MultiplyAddWorkChain = WorkflowFactory('arithmetic.multiply_add')
-       ...: builder = MultiplyAddWorkChain.get_builder()
-
-To explore the inputs of the work chain, you can use tab autocompletion by typing ``builder.`` and then hitting ``TAB``.
-If you want to get more details on a specific input, you can simply add a ``?`` and press enter:
+.. code-block:: console
 
-.. code-block:: ipython
+    $ verdi process status 164
+    MultiplyAddIsEvenWorkChain<164> Finished [0] [1:is_even]
+        ├── MultiplyAddWorkChain<165> Finished [0] [3:result]
+        │   ├── multiply<166> Finished [0]
+        │   └── ArithmeticAddCalculation<168> Finished [0]
+        └── is_even<172> Finished [0]
 
-    In [2]: builder.x?
-    Type:        property
-    String form: <property object at 0x119ad2dd0>
-    Docstring:   {"name": "x", "required": "True", "valid_type": "<class 'aiida.orm.nodes.data.int.Int'>", "non_db": "False"}
+Note that this command also recursively shows the processes called by the subprocesses of the ``MultiplyAddIsEvenWorkChain`` work chain.
 
-Here you can see that the ``x`` input is required, needs to be of the ``Int`` type and is stored in the database (``"non_db": "False"``).
+As mentioned earlier, you can also expose the outputs of the ``MultiplyAddWorkChain`` using the :meth:`~plumpy.ProcessSpec.expose_outputs` method.
+Say we want to add the ``result`` of the ``MultiplyAddWorkChain`` as one of the outputs of the extended work chain:
 
-Using the builder, the inputs of the ``WorkChain`` can be provided one by one:
-
-.. code-block:: ipython
-
-    In [3]: builder.code = load_code(label='add')
-       ...: builder.x = Int(2)
-       ...: builder.y = Int(3)
-       ...: builder.z = Int(5)
+.. code-block:: python
 
-Once the *required* inputs of the workflow have been provided to the builder, you can either run the work chain or submit it to the daemon:
+    @classmethod
+    def define(cls, spec):
+        """Specify inputs and outputs."""
+        super().define(spec)
+        spec.expose_inputs(MultiplyAddWorkChain, namespace='multiply_add')
+        spec.outline(
+            cls.multiply_add,
+            cls.is_even,
+        )
+        spec.expose_outputs(MultiplyAddWorkChain)
+        spec.output('is_even', valid_type=Bool)
+
+Since there is not one output port that is shared by all process classes, it is less critical to use the ``namespace`` argument when exposing outputs.
```

Comment:
Hmm, I'm not sure if we should recommend it, but an extra sentence as a warning is definitely welcome.

Reply:
Yeah, now that I think about it maybe you are right, you would tipically handpick the results of the workchain rather than just exposing them directly. Even in complex workchains you won't want to just vomit every output produced in the middle, they can anyways be accessed by exploring the graph (unlike inputs which you need to expose so that users can customize the runs).

Feel free to disregard this, or maybe briefly express this idea I rambled in the previous paragraph.